pkg/webhook/resources/schedulevmbackup/validator.go

```diff
@@ -0,0 +1,203 @@
+	"github.com/harvester/harvester/pkg/webhook/types"
+)
+
+const (
+	fieldCron       = "spec.cron"
+	fieldMaxFailure = "spec.maxFailure"
+	fieldSuspend    = "spec.suspend"
+	fieldVMBackup   = "spec.vmbackup"
+
+	minCronGranularity = time.Hour
+	minCronOffset      = 10 * time.Minute
+)
+
+type scheuldeVMBackupValidator struct {
+	types.DefaultValidator
+	settingCache   ctlharvesterv1.SettingCache
+	secretCache    ctlv1.SecretCache
+	svmbackupCache ctlharvesterv1.ScheduleVMBackupCache
+}
+
+func NewValidator(
+	settingCache ctlharvesterv1.SettingCache,
+	secretCache ctlv1.SecretCache,
+	svmbackupCache ctlharvesterv1.ScheduleVMBackupCache,
+) types.Validator {
+	return &scheuldeVMBackupValidator{
+		settingCache:   settingCache,
+		secretCache:    secretCache,
+		svmbackupCache: svmbackupCache,
+	}
+}
+
+func (v *scheuldeVMBackupValidator) Resource() types.Resource {
+	return types.Resource{
+		Names:      []string{v1beta1.ScheduleVMBackupResourceName},
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   v1beta1.SchemeGroupVersion.Group,
+		APIVersion: v1beta1.SchemeGroupVersion.Version,
+		ObjectType: &v1beta1.ScheduleVMBackup{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+			admissionregv1.Update,
+			admissionregv1.Delete,
+		},
+	}
+}
+
+func (v *scheuldeVMBackupValidator) checkTargetHealth() error {
+	targetSetting, err := v.settingCache.Get(settings.BackupTargetSettingName)
+	if err != nil {
+		return err
+	}
+
+	if targetSetting.Value == "" {
+		return fmt.Errorf("setting %s is empty", settings.BackupTargetSettingName)
+	}
+
+	target, err := settings.DecodeBackupTarget(targetSetting.Value)
+	if err != nil {
+		return fmt.Errorf("can't decode setting %s value %s, error: %w", settings.BackupTargetSettingName, targetSetting.Value, err)
+	}
+
+	if target.IsDefaultBackupTarget() {
+		return fmt.Errorf("setting %s is not set", settings.BackupTargetSettingName)
+	}
+
+	if _, err := util.GetBackupStoreDriver(v.secretCache, target); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func cronGranularityCheck(v *scheuldeVMBackupValidator, svmbackup *v1beta1.ScheduleVMBackup) error {
+	granularity, err := util.GetCronGranularity(svmbackup)
+	if err != nil {
+		return werror.NewInvalidError("invalid cron format", fieldCron)
+	}
+
+	if granularity < minCronGranularity {
+		return fmt.Errorf("schedule granularity %s less than %s", granularity.String(), minCronGranularity.String())
+	}
+
+	svmbackups, err := v.svmbackupCache.GetByIndex(indexeres.ScheduleVMBackupByCronGranularity, granularity.String())
+	if err != nil {
+		return err
+	}
+
+	for _, s := range svmbackups {
+		// skip check with svmbackup itself
+		if s.UID == svmbackup.UID {
+			continue
+		}
+
+		offset, err := util.CalculateCronOffset(svmbackup, s)
+		if err != nil {
+			return err
+		}
+
+		if offset < minCronOffset {
+			return fmt.Errorf("same granularity as schedule %s and offset less than %s", s.Name, minCronOffset.String())
```

Comment:
```suggestion
			return fmt.Errorf("same granularity as schedule %s and offset less than %s minutes", s.Name, minCronOffset.String())
```

just a suggestion to make the error message more understandable.

Reply:
Hi @bk201 
the error message will be like this `The request is invalid: spec.cron: same granularity as schedule svmbackup-vm1 and offset less than 10m0s`, maybe we don't need to add the additional word `minutes` in the message?

Reply:
no need my bad, I was not aware it's a time type.